Python/bltinmodule.c

```diff
@@ -2737,6 +2739,496 @@ builtin_issubclass_impl(PyObject *module, PyObject *cls,
     return PyBool_FromLong(retval);
 }
 
+/*[clinic input]
+isimmutable as builtin_isimmutable
+
+    obj: object
+    /
+
+Return whether 'obj' is immutable.
+[clinic start generated code]*/
+
+static PyObject *
+builtin_isimmutable(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=80c746a3bd7adb46 input=15c0c9d2da47bc15]*/
+{
+    _Py_VPYDBG("isimmutable(");
+    _Py_VPYDBGPRINT(obj);
+    _Py_VPYDBG(") region: %lu\n", Py_REGION(obj));
+    return PyBool_FromLong(_Py_IsImmutable(obj));
+}
+
+typedef struct node_s {
+    PyObject* object;
+    struct node_s* next;
+} node;
+
+typedef struct stack_s {
+    node* head;
+} stack;
+
+stack* stack_new(void){
+    stack* s = (stack*)malloc(sizeof(stack));
+    if(s == NULL){
+        return NULL;
+    }
+
+    s->head = NULL;
+
+    return s;
+}
+
+bool stack_push(stack* s, PyObject* object){
+    node* n = (node*)malloc(sizeof(node));
+    if(n == NULL){
+        return true;
+    }
+
+    n->object = object;
+    Py_INCREF(object);
+    n->next = s->head;
+    s->head = n;
+    return false;
+}
+
+PyObject* stack_pop(stack* s){
+    if(s->head == NULL){
+        return NULL;
+    }
+
+    node* n = s->head;
+    PyObject* object = n->object;
+    s->head = n->next;
+    free(n);
+    Py_DECREF(object);
+
+    return object;
+}
```

Comment:
I think this should be optimised in a future PR.

Reply:
Agreed.